tests/Doctrine/Tests/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -34,6 +36,19 @@ public function testGetCacheDriver()
         self::assertSame($cache, $this->cmf->getCacheDriver());
     }
 
+    public function testGetMetadataForAnonymousClass()
+    {
+        $object = new class () extends stdClass {
+        };
+
+        try {
+            $this->cmf->getMetadataFor(get_class($object));
+            self::fail('must throw MappingException error');
+        } catch (MappingException $error) {
+            self::assertEquals(1, preg_match("/^The class '([^']+)' is anonymous.$/", $error->getMessage()));
+        }
```

Comment:
Let's make the test more simple and robust:

```suggestion
        $this->expectException(MappingException::class);
        $this->expectExceptionMessage('anonymous');
        $this->cmf->getMetadataFor(get_class($object));
```

Reply:
Thank. Fixed. I think what `expectExceptionMessage` full match message - `$error->getMessage() === $expectedMessage`, but I see what `ExceptionMessage Constraint` match by `strpos`.